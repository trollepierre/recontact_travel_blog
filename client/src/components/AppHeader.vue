<template>
  <div class="app-header">
    <header class="page__header">
      <div class="page__container page__header--container">
        <a class="logo-link" href="/">
          <span class="logo-link__job">Recontact</span>
          <span class="logo-link__board">Me</span>
        </a>
        <nav class="app-header__navigation navigation" role="navigation" aria-label="site navigation">
          <ol class="navigation__links">
            <li class="navigation__link">
              <button class="navbar-action navbar-action__sync" type="button" :disabled="isClicked" @click.prevent.once="synchronise">Synchroniser</button>
            </li>
          </ol>
        </nav>
      </div>
    </header>
  </div>
</template>
<script>
  import syncApi from '@/api/sync';
  import notificationService from '@/services/notification';

  export default {
    name: 'AppHeader',
    data() {
      return {
        isClicked: false,
      };
    },
    methods: {
      disableButton() {
        this.isClicked = true;
      },

      synchronise() {
        this.disableButton();
        syncApi.launch()
          .then((response) => {
            const message = 'La synchronisation s\'est effectuée sans problème !';
            notificationService.success(this, message);
            console.log(response);
          })
          .catch((err) => {
            const message = `Erreur : Problème durant la synchronisation : ${err.message}`;
            notificationService.error(this, message);
            console.log(err);
          });
      },
    },
  };
</script>

<style scoped>
  .page__header {
    height: 60px;
    background: #ffffff;
    border-bottom: 1px solid #e6e6e6;
    width: 100%;
    padding-left: 0;
    position: fixed;
    top: 0;
  }

  .page__header--container {
    display: flex;
    justify-content: space-between;
  }

  .logo-link {
    text-decoration: none;
    font-size: 26px;
    font-weight: 900;
    display: inline-block;
    padding: 15px 0;
  }

  .logo-link__job {
    color: #07c;
  }

  .logo-link__board {
    color: #F48024;
  }

  .logout-link {
    color: #9199a1;
    display: inline-block;
    padding: 17px 0;
    line-height: 28px;
    text-decoration: none;
  }

  .logout-link:hover {
    text-decoration: underline;
  }

  .page__header {
    height: 60px;
    background: #ffffff;
    border-bottom: 1px solid #e6e6e6;
    width: 100%;
    position: fixed;
    top: 0;
  }

  .page__header--container {
    display: flex;
    justify-content: center;
  }

  .logo-link {
    text-decoration: none;
    font-size: 26px;
    display: inline-block;
    padding: 15px 0;
  }

  .logo-link__job {
    color: #07c;
  }

  .logo-link__board {
    color: #d14800;
    font-weight: 900;
  }

  .navbar-action {
    cursor: pointer;
    background: transparent;
    font-size: 16px;
    border: none;
    padding: 16px 0;
    line-height: 28px;
    color: #333333;
    display: inline-block;
  }

  .navbar-action:hover {
    text-decoration: underline;
  }

  .navigation {
    display: none;
  }

  .navigation__links {
    list-style: none;
    margin: 0;
    display: inline-flex;
  }

  .navigation__link {
    margin-left: 25px;
  }

  .page__container {
    margin: 0 auto;
  }

  @media only screen and (min-width: 640px) {
    .page__header--container {
      justify-content: space-between;
    }

    .app-header__navigation {
      display: inline-block;
    }
  }
</style>
