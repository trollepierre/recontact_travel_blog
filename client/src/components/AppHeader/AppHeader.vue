<template>
  <div class="app-header">
    <header class="page__header">
      <div class="page__container page__header--container">
        <a
          :title="home"
          class="logo-link"
          href="/">
          <span class="logo-link__recontact">
            Recontact
          </span>
          <span class="logo-link__me">
            Me
          </span>
        </a>
        <nav
          class="app-header__navigation navigation"
          role="navigation"
          aria-label="site navigation">
          <ol class="navigation__links">
            <li class="navigation__link">
              <button
                class="navbar-action navbar-action__subscribe"
                type="button"
                @click.prevent="displaySubscribeModal">
                {{ $t("subscribe") }}
              </button>
            </li>
            <li class="navigation__link">
              <button
                class="navbar-action navbar-action__suggestion"
                type="button"
                @click.prevent="displayFeedbackModal">
                {{ $t("suggestion") }}
              </button>
            </li>
            <li class="navigation__link tdm">
              <a
                :title="tdm"
                class="navbar-action navbar-action__tdm"
                href="http://worldtour.recontact.me">
                <img
                  class="tdm__image"
                  :alt="logo"
                  src="/static/tdm.jpg">
              </a>
            </li>
          </ol>
        </nav>
      </div>
    </header>
  </div>
</template>
<script>
  export default {
    name: 'AppHeader',
    computed: {
      tdm() {
        return this.$t('tdm')
      },
      home() {
        return this.$t('home')
      },
      logo() {
        return this.$t('logo')
      },
    },
    methods: {
      displaySubscribeModal() {
        this.$modal.show('subscribe-modal')
      },

      displayFeedbackModal() {
        this.$modal.show('feedback-modal')
      },
    },
    i18n: {
      messages: {
        fr: {
          subscribe: 'S’abonner',
          suggestion: 'Laisser un message',
          problem: 'Un problème ?',
          tdm: 'Retrouver l’ancien site du tour du monde de Pierre et Benoît',
          home: 'Page d’accueil',
          logo: 'Logo du site',
        },
        en: {
          subscribe: 'Subscribe',
          suggestion: 'Leave a message',
          problem: 'A problem?',
          tdm: 'Go to see the former website of the world trip of Pierre and Benoît',
          home: 'Home page',
          logo: 'Logo of the site',
        },
      },
    },
  }
</script>

<style scoped>
  .page__header {
    height: 60px;
    background: #ffffff;
    border-bottom: 1px solid #e6e6e6;
    width: 100%;
    padding-left: 0;
    border-bottom: 1px solid #e6e6e6;
  }

  .page__header--container {
    display: flex;
    justify-content: center;
  }

  .logo-link {
    text-decoration: none;
    font-size: 26px;
    font-weight: 900;
    display: inline-block;
    padding: 15px 0;
  }

  .logo-link:hover {
    outline: -webkit-focus-ring-color auto 5px;
  }

  .logo-link__recontact {
    color: #07c;
    padding-left: 10px;
  }

  .logo-link__me {
    color: #F48024;
  }

  .navbar-action {
    cursor: pointer;
    background: transparent;
    font-size: 16px;
    border: none;
    padding: 16px 0;
    line-height: 28px;
    color: #333333;
    display: inline-block;
  }

  .navbar-action:hover {
    text-decoration: underline;
    outline: -webkit-focus-ring-color auto 5px;
  }

  .navbar-action__suggestion, .navbar-action__tdm {
    display: none;
  }

  .navigation__links {
    list-style: none;
    margin: 0;
    display: inline-flex;
  }

  .navigation__link {
    margin-left: 25px;
  }

  .page__container {
    margin: 0 auto;
  }

  .tdm {
    display: inline-flex;
  }

  .navbar-action__tdm {
    padding: 10px;
  }

  @media only screen and (min-width: 640px) {
    .page__header--container {
      justify-content: space-between;
    }

    .navbar-action__suggestion, .navbar-action__tdm {
      display: inline-block;
    }

    .logo-link__recontact {
      color: #07c;
      padding-left: 0;
    }
  }
</style>
