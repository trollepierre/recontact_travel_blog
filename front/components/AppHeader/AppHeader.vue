<template>
  <header
    class="header"
    :class="{ 'navbar-hidden': !showNavbar }">
    <div class="container">
      <a
        :title="$t('home')"
        class="logo"
        href="/">
        <img
          :alt="$t('logo')"
          class="icon"
          width="38"
          height="38"
          src="../../static/velo.png">
        <span class="recontact">Recontact</span>
        <span class="me">Me</span>
      </a>
      <nav
        v-if="isArticlePage"
        aria-label="navigation">
        <ul class="navigation">
          <li class="article">
            <app-button
              v-if="previousArticleId"
              :to="previousArticleId"
              tag="nuxt-link"
              class="article"
              :text="$t('previousArticle')"/>
          </li>
          <li class="article-text">
            <p class="id">
              Article {{ articleId }}
            </p>
          </li>
          <li class="article">
            <app-button
              v-if="nextArticleId"
              :to="nextArticleId"
              tag="nuxt-link"
              class="article"
              :text="$t('nextArticle')"/>
          </li>
        </ul>
      </nav>
      <nav aria-label="navigation">
        <nav-control/>
      </nav>
    </div>
  </header>
</template>
<script>
/* eslint-disable  max-lines */
  import AppButton from '@/components/AppButton/AppButton'
  import NavControl from '@/components/AppHeader/NavControl/NavControl'

  export default {
    name: 'AppHeader',
    components: { NavControl, AppButton },
    data: () => ({
      otherUrl: '',
      showNavbar: true,
      lastScrollPosition: 0,
      isArticlePage: false,
      articleId: null,
      previousArticleId: null,
      nextArticleId: null,
    }),
    mounted() {
      window.addEventListener('scroll', this.onScroll)
      this.isArticlePage = window.location.pathname.includes('/articles/')
      console.log('this.isArticlePage:', this.isArticlePage)
      if (this.isArticlePage) {
        // eslint-disable-next-line prefer-destructuring
        this.articleId = window.location.pathname.split('/articles/')[1].split('/')[0]
        this.previousArticleId = this.articleId !== '1' ? `/articles/${this.articleId - 1}` : null
        console.log('this.articleId:', this.articleId)
        this.nextArticleId = `/articles/${this.articleId - 1 + 2}`
        console.log('this.nextArticleId:', this.nextArticleId)
      }
    },
    beforeDestroy() {
      window.removeEventListener('scroll', this.onScroll)
    },
    methods: {
      onScroll() {
        // Get the current scroll position
        const currentScrollPosition = window.pageYOffset || document.documentElement.scrollTop
        // Because of momentum scrolling on mobiles, we shouldn't continue if it is less than zero
        if (currentScrollPosition < 0) {
          return
        }
        // Stop executing this function if the difference between
        // current scroll position and last scroll position is less than some offset
        if (Math.abs(currentScrollPosition - this.lastScrollPosition) < 60) {
          return
        }
        // Here we determine whether we need to show or hide the navbar
        this.showNavbar = currentScrollPosition < this.lastScrollPosition // Set the current scroll position as the last scroll position
        this.lastScrollPosition = currentScrollPosition
      },
    },
    i18n: {
      silentTranslationWarn: true,
      messages: {
        fr: {
          home: 'Page dâ€™accueil',
          logo: 'Logo',
          previousArticle: '<',
          nextArticle: '>',
          article: 'Article',
        },
        en: {
          home: 'Home page',
          logo: 'Logo',
          previousArticle: '<',
          nextArticle: '>',
          article: 'Article',

        },
      },
    },
  }
</script>

<style lang="scss" scoped>
.header {
  height: 60px;
  background: $header-bg;
  border-bottom: 1px solid $border;
  width: 100%;
  padding-left: 0;
  position: fixed;
  z-index: 1;
}

.navbar-hidden {
  transform: translate3d(0, -100%, 0);
}

.container {
  height: 100%;
  display: flex;
  margin: 0 auto;
  justify-content: space-around;
}

.logo {
  text-decoration: none;
  font-size: 20px;
  font-weight: 900;
  display: flex;
  justify-content: space-between;
  align-items: center;
  font-family: serif;
  border-radius: 4px;
}

.recontact {
  color: $logo-primary;
  padding-top: 5px;
  padding-left: 5px;
  padding-bottom: 0;
  align-self: center;
}

.me {
  align-self: center;
  padding: 5px 5px 0;
  color: $logo-secondary;
}

.icon {
  display: block;
  height: 38px;
  margin-left: -40px;
  font-size: 8px;
  color: $logo-color;
}

.navigation {
  list-style: none;
  margin: 0;
  display: inline-flex;
  padding-left: 0;
  height: 100%;
}

.article, .article-text {
  font-family: serif;
  padding-left: 10px;
  color: $nav-color;
}

.id {
  margin-top: 20px;
}

.button.article { // used in [<] Article 85 [>]
  padding: 0 10px;
}

.button {
  background-color: $header-bg;
}

.button:hover {
  background-color: $button-hover;
}

.article, .article-text {
  display: flex;
  align-items: center;
  justify-content: center;
}

.logo:hover {
  border: 1px solid $logo-hover;
  padding-left: 50px;
  margin-left: -50px;
  border-radius: 10px;
}

@media only screen and (max-width: 400px) {
  .icon {
    display: none;
  }

  .logo:hover {
    padding-left: 0;
    margin-left: 0;
  }
}

@media only screen and (min-width: 640px) {
  .container {
    max-width: 500px;
    justify-content: space-between;
  }

  .recontact {
    padding-left: 8px;
  }

  .me {
    padding-right: 10px;
  }
}

@media only screen and (min-width: 992px) {
  .container {
    max-width: 716px;
  }

  .logo {
    font-size: 26px;
  }

  .logo {
    padding: 10px 0;
    margin: 0;
    border: none;
    border-radius: 0;
  }

  .article, .article-text {
    font-size: 18px;
  }

  .button.article {
    font-size: 14px;
  }
}

@media only screen and (min-width: 1200px) {
  .container {
    max-width: 888px;
  }
}
</style>
