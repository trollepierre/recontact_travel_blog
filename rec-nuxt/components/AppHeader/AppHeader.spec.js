import VueI18n from 'vue-i18n'
import VueModal from 'vue-js-modal'
import AppHeader from './AppHeader.vue'

describe('Component | AppHeader.vue', () => {
  let localVue
  let wrapper

  beforeEach(() => {
    localVue = createLocalVue()
    localVue.use(VueI18n)
  })

  it('should be named "AppHeader"', () => {
    wrapper = shallowMount(AppHeader, { localVue })

    expect(wrapper.name()).toEqual('AppHeader')
  })

  describe('template', () => {
    it('should match snapshot', () => {
      wrapper = shallowMount(AppHeader, { localVue })

      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('methods', () => {
    beforeEach(() => {
      wrapper = shallowMount(AppHeader, { localVue })
      localVue.use(VueModal)
    })

    describe('#displayFeedbackModal', () => {
      it('should display the feedback-modal', () => {
        wrapper.vm.$modal.show = jest.fn()

        wrapper.vm.displayFeedbackModal()

        expect(wrapper.vm.$modal.show).toHaveBeenCalledWith('feedback-modal')
      })
    })

    describe('#displaySubscribeModal', () => {
      it('should display the subscribe-modal', () => {
        wrapper.vm.$modal.show = jest.fn()

        wrapper.vm.displaySubscribeModal()

        expect(wrapper.vm.$modal.show).toHaveBeenCalledWith('subscribe-modal')
      })
    })
  })

  describe('events', () => {
    describe('clicking on button "Laisser un message"', () => {
      it('should call displayFeedbackModal', () => {
        wrapper.vm.displayFeedbackModal = jest.fn()

        wrapper.find('button.navbar-action.navbar-action__suggestion').trigger('click')

        expect(wrapper.vm.displayFeedbackModal).toHaveBeenCalled()
      })
    })

    describe('clicking on button "S\'abonner"', () => {
      it('should call displaySubscribeModal', () => {
        wrapper.vm.displaySubscribeModal = jest.fn()

        wrapper.find('button.navbar-action.navbar-action__subscribe').trigger('click')

        expect(wrapper.vm.displaySubscribeModal).toHaveBeenCalled()
      })
    })
  })

  describe('locales', () => {
    const languages = Object.keys(AppHeader.i18n.messages)

    it('contains 2 languages', () => {
      expect(languages).toHaveLength(2)
      expect(languages).toEqual(['fr', 'en'])
    })

    describe('each language', () => {
      describe('fr', () => {
        const locales = Object.keys(AppHeader.i18n.messages.fr)

        it('contains 6 locales', () => {
          expect(locales).toHaveLength(6)
          expect(locales).toEqual(['subscribe', 'suggestion', 'problem', 'tdm', 'home', 'logo'])
        })
      })

      describe('en', () => {
        const locales = Object.keys(AppHeader.i18n.messages.en)

        it('contains 6 locales', () => {
          expect(locales).toHaveLength(6)
          expect(locales).toEqual(['subscribe', 'suggestion', 'problem', 'tdm', 'home', 'logo'])
        })
      })
    })
  })
})
