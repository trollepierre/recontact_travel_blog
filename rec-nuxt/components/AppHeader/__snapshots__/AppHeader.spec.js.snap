// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component | AppHeader.vue template should match snapshot 1`] = `
<div
  class="app-header"
>
  <header
    class="page__header"
  >
    <div
      class="page__container page__header--container"
    >
      <a
        class="logo-link"
        href="/"
        title="home"
      >
        <span
          class="logo-link__recontact"
        >
          Recontact
        </span>
         
        <span
          class="logo-link__me"
        >
          Me
        </span>
      </a>
       
      <nav
        aria-label="site navigation"
        class="app-header__navigation navigation"
        role="navigation"
      >
        <ol
          class="navigation__links"
        >
          <!---->
           
          <!---->
           
          <!---->
           
          <li
            class="navigation__link other-language"
          >
            <button
              class="navbar-action navbar-action__other-language"
              type="button"
            >
              
              otherLanguage
            
            </button>
          </li>
        </ol>
      </nav>
    </div>
  </header>
</div>
`;
